edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +223,23 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let lock = Arc::new(AtomicU32::new(0));
```

Comment:
It would be better to do what Gordon did in his PR https://github.com/Azure/iot-identity-service/pull/249/files

Specifically, start the lock at the upper limit instead of 0, and use `fetch_update` to optionally decrement it and check whether decrementing succeeded in the same operation, instead of what you're doing right now by always `fetch_add`ing and then `fetch_sub`ing if it exceeds the limit.

Reply:
Ok. I know the > 10 check is not atomic. I didn't bother much because, anyway the >10 is kinda arbitrary and also we are already in the task when we perform the check.
But it doesn't hurt, I fixed it.